src/main/java/study/StringSum.java

```diff
@@ -0,0 +1,55 @@
+package study;
+
+import study.validator.StringSumValidator;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringSum {
+
+    private static final String DEFAULT_DELIMITER = ",|:";
+    private static final String CUSTOM_DELIMITER_START = "//";
+    private static final int DEFAULT_NUMBER = 0;
+    private static final Pattern CUSTOM_DELIMITER_PATTERN = Pattern.compile("//(.)\n");
+    private static final int NEXT_LINE_INDEX = 1;
+
+
+    public int sumStringByDelimiter(String data) {
+
+        if (!StringSumValidator.validateBlank(data)) {
+            return DEFAULT_NUMBER;
+        }
+
+        String[] numbers = splitData(data);
+        return sumNumbers(numbers);
+    }
+
+    private String[] splitData(String data) {
+
+        String delimiter = DEFAULT_DELIMITER;
+
+        if (data.startsWith(CUSTOM_DELIMITER_START)) {
+            delimiter = extractCustomDelimiter(data);
+            data = data.substring(data.indexOf('\n') + NEXT_LINE_INDEX);
+        }
+        return data.split(delimiter);
+    }
+
+    private String extractCustomDelimiter(String data) {
+
+        Matcher m = CUSTOM_DELIMITER_PATTERN.matcher(data);
+        if (m.find()) {
+            return Pattern.quote(m.group(1));
```

Comment:
```suggestion
            int demiDelimiterIndex = 1;
            return Pattern.quote(m.group(demiDelimiterIndex));
```

요렇게 하면, 동료개발자가 봤을때 빠르게 이해할수 있을것 같기도 하네요 😄

Reply:
매직 넘버를 없이 하는게 목표였는데 놓친 부분이 있었네요 😢 
다음부터 더 신경써서 하겠습니다 감사합니다!